specifyweb/frontend/js_src/lib/components/FormSliders/AllAttachments.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { AnySchema } from '../DataModel/helperTypes';
+import { Collection } from '../DataModel/specifyModel';
+import { SpecifyResource } from '../DataModel/legacyTypes';
+import { RA } from '../../utils/types';
+import { useAttachment } from '../Attachments/Plugin';
+import { AttachmentViewer } from '../Attachments/Viewer';
+import { useTriggerState } from '../../hooks/useTriggerState';
+import { Dialog } from '../Molecules/Dialog';
+import { Button } from '../Atoms/Button';
+import { commonText } from '../../localization/common';
+import { attachmentsText } from '../../localization/attachments';
+import { useBooleanState } from '../../hooks/useBooleanState';
+
+export function AttachmentsCollection<SCHEMA extends AnySchema>({
```

Comment:
Can you please remind me why attachment gallery wasn't used for this?
Seems like it would be provide the same functionality - in fact, didn't you already use it on the branch where you added ability to see attachments belonging to a record set?

Reply:
Because I preferred creating a new component with only what I needed. For the attachments in RecordSet I also created a new component.

Reply:
> Because I preferred creating a new component with only what I needed. For the attachments in RecordSet I also created a new component.

😢 

This way you might be duplicating some code (thus 3 places to maintain and 3 places that could have a bug). Also, if there is a bug in one of the places, fixing it won't fix the other 3. That means if doing testing, all 3 places have to be tested - if the same component was reused, testing just one would most likely be sufficient.

Can you please look at using `<AttachmentGallery />` here and in record sets? Looking at the props it accepts, you should just be able to use it without any extra modifications, or did you run into some issues? We can also meet and try this together if you want

Reply:
I'd be happy to read you suggested code implementing attachmentGallery. How would get RA<SerializedResource<Attachment>> to pass as attachments prop to Gallery in IntegratedRecordSelector.tsx?

Reply:
Your new compoment accepts a Collection<...>, right? If you have a collection you can get RA<...> just like that:

```
collection.models
```